Examples/Algorithms/Vertexing/src/AdaptiveMultiVertexFinderAlgorithm.cpp

```diff
@@ -74,22 +81,46 @@ AdaptiveMultiVertexFinderAlgorithm::AdaptiveMultiVertexFinderAlgorithm(
   if (m_cfg.outputVertices.empty()) {
     throw std::invalid_argument("Missing output vertices collection");
   }
+  if (m_cfg.seedFinder == SeedFinder::TruthSeeder &&
+      m_cfg.inputTruthVertices.empty()) {
+    throw std::invalid_argument("Missing input truth vertex collection");
+  }
+
+  // Sanitize the configuration
+  if (m_cfg.seedFinder != SeedFinder::TruthSeeder &&
+      (!m_cfg.inputTruthParticles.empty() ||
+       !m_cfg.inputTruthVertices.empty())) {
+    ACTS_INFO("Ignoring truth input as seed finder is not TruthSeeder");
+    m_cfg.inputTruthVertices.clear();
+    m_cfg.inputTruthVertices.clear();
+  }
 
   m_inputTrackParameters.initialize(m_cfg.inputTrackParameters);
+  m_inputTruthParticles.maybeInitialize(m_cfg.inputTruthParticles);
+  m_inputTruthVertices.maybeInitialize(m_cfg.inputTruthVertices);
   m_outputProtoVertices.initialize(m_cfg.outputProtoVertices);
   m_outputVertices.initialize(m_cfg.outputVertices);
 }
 
-auto AdaptiveMultiVertexFinderAlgorithm::makeVertexFinder() const
-    -> Acts::AdaptiveMultiVertexFinder {
-  std::shared_ptr<const Acts::IVertexFinder> seedFinder;
+std::unique_ptr<Acts::IVertexFinder>
+AdaptiveMultiVertexFinderAlgorithm::makeVertexSeeder() const {
+  if (m_cfg.seedFinder == SeedFinder::TruthSeeder) {
+    using Seeder = ActsExamples::TruthVertexSeeder;
+    Seeder::Config seederConfig;
+    seederConfig.useXY = false;
```

Comment:
Was this hard-coded before as well?

Reply:
I also hard coded this in the first version. Since all our other seeder only do `Z` and `T` using `X` and `Y` too might be too optimistic for a comparison with the new truth seeder